File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/typescript/evm/policies/createUSDSpendRestrictionPolicy.ts
// Usage: pnpm tsx evm/policies/createUSDSpendRestrictionPolicy.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const cdp = new CdpClient();
const policy = await cdp.policies.createPolicy({
  policy: {
    scope: "account",
    description: "Reject over one hundred dollars",
    rules: [
      {
        action: "reject",
        operation: "sendEvmTransaction",
        criteria: [
          {
            type: "netUSDChange",
            changeCents: 10000,
            operator: ">",
          },
        ],
      },
    ],
  },
});
console.log("Created USD restriction policy: ", policy.id);

File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/python/evm/policies/create_usd_spend_restriction_policy.py
# Usage: uv run python evm/policies/create_usd_spend_restriction_policy.py

import asyncio

from cdp import CdpClient
from dotenv import load_dotenv
from cdp.policies.types import (
    CreatePolicyOptions,
    SendEvmTransactionRule,
    NetUSDChangeCriterion,
)

load_dotenv()

async def main():
    async with CdpClient() as cdp:
        # Create a policy that allows sending up to 100 dollars worth of assets at a time
        policy = CreatePolicyOptions(
            scope="account",
            description="Accept up to 100 dollars",
            rules=[
                SendEvmTransactionRule(
                    action="accept",
                    criteria=[
                        NetUSDChangeCriterion(
                            changeCents=10000,
                            operator="<",
                        ),
                    ],
                ),
            ],
        )

        result = await cdp.policies.create_policy(policy=policy)

        print("Created USD spend restrictions policy: ", result.id)


asyncio.run(main())
